### ISSUE TITLE
Alfred Maestro v2 Meta Issue

### ISSUE THREAD

**Issue by iansinnott** [iansinnott/alfred-maestro#13]
Let's figure out what new changes would be helpful in the next version! This lib has been _exactly_ what I myself needed for a while now so I haven't given a whole lot of thought to possible improvements. But @nikitavoloboev has made some good suggestions in the past and I think aggregating all suggestions together here will help bring them to fruition. 

**IMPORTANT NOTE:** The feasibility of any of these features depends entirely on the [KM scripting API](https://wiki.keyboardmaestro.com/manual/Scripting). So some features might actually be impossible to implement without changes to KM itself.

So, with that said, let's talk features. Here are some ideas that have been raised in the past:

* #12 Open the macro in Keyboard Maestro editor
* #10 Show the folder where the macro is in in Subtitle
* #8 Show hotkey triggers in the subtext of each macro

Furthermore, I wrote this lib a long time ago when PHP was my primary language but that has since changed and probably hinders contributions to this library. So, I'm considering a full rewrite in either **Python** or **JavaScript** (Sidenote: this whole lib is less than 200 lines of code, so a full rewrite shouldn't be too onerous).

I'm interested in feedback on all of this and if anyone has ideas on other features that would make this lib more useful please add them!

**Comment by nikitavoloboev** [comment]
Can I consider trying to rewrite this in Go using [AwGo](https://github.com/deanishe/awgo)?

**Comment by iansinnott** [comment]
@nikitavoloboev Hm.. Go wouldn't be my first choice. Nothing against the language, but one thing I'd very much like to accomplish with a rewrite is to make this lib more accessible so people can contribute. I'm not sure go is as accessible or widely used as JS or Python. Thoughts?

Also, can a go lib be compiled to a standalone binary? A key requirement is that whatever we choose it runs on Macs without installing anything.

**Comment by nikitavoloboev** [comment]
> Also, can a go lib be compiled to a standalone binary? A key requirement is that whatever we choose it runs on Macs without installing anything.

Yep. That's its biggest advantage. It compiles to one binary and is super fast. And AwGo is really impressive in what it allows you to do.

> make this lib more accessible so people can contribute

I think Go and Python are two great choices. I just prefer Go as I want the workflows to be running as fast as possible. And we can make contributions approachable by having clear documentation and ways in which to bootstrap and hack on the workflow. Which we can do easily.

**Comment by nikitavoloboev** [comment]
Also we can use the amazing work done by Dan that made this: 

https://forum.keyboardmaestro.com/t/macro-execute-trigger-macro-by-name-spotlight/4667 

To search for macros to execute.

**Comment by nikitavoloboev** [comment]
And another thing I really need in my life is searching for KM macro groups to jump to. This is possible to do.

**Comment by iansinnott** [comment]
Well... speed is indeed a great feature. There's definitely noticeable lag sometimes with the current implementation. If using go could reduce that lag below perceptibility that would be a big win.

I still have concerns about the accessibility but as you say we can probably alleviate a lot of the friction there. So what the hell—if you wan't to lead a rewrite to Go then i'm on board 🚀

**Comment by nikitavoloboev** [comment]
I will try to read through the PHP code there and see if I can make sense of it and will try get something working.

**Comment by iansinnott** [comment]
Sounds great. Be sure to ignore `workflows.php` since that's just the equivalent of AwGo. Other than that:

* `km.scpt`: Grab all macros from KM (result is XML string)
* `main.php`: Parse XML string and echo out workflow-formatted results for alfred to display

**Comment by maltsev** [comment]
I think we can close this issue and continue the discussion about features in their separate issues.

@iansinnott you can also remove your [dev branch](https://github.com/iansinnott/alfred-maestro/tree/dev) since it's outdated and no longer needed.

**Comment by nikitavoloboev** [comment]
I agree that this should be closed.

Also @maltsev I think the `make` script can be replaced by using [this tool](https://godoc.org/github.com/jason0x43/go-alfred/alfred). I use it all the time to make my Go workflows and love it.

**Comment by maltsev** [comment]
> Also @maltsev I think the make script can be replaced by using this tool

It seems that this tool doesn't support test running. So we should at least keep _make test_. In general, I think _make_ is a good way for running build scripts. Instead of remembering how to run tests and build each project all I have to remember is _make test_ and _make build_. Anyway, I've created #16 for discussing that.
